### PR TITLE
Step 6 filter devices for platform

### PR DESCRIPTION
**Pull request by godrei**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _PATCH_ [version update](https://semver.org/)

### Context

The step ensures all the Bitrise test devices are registered on the Apple Developer Portal and ensures the test devices are included in the Step-managed Provisioning Profile.

Bitrise test devices can contain any type of device (macOS, tvOS, ....), however not all types of devices can be used in each Profile type (e.g.: iOS Profile can not be used to provision a tvOS app). 

The Step did not filter the compatible devices, which can result in false-negative Profile checks (e.g.: a tvOS Xcode managed Profile was marked as invalid because it did not include an iPhone test device registered on Bitrise).

Resolves: https://bitrise.atlassian.net/browse/STEP-6

### Changes

- **Filter test devices based on the project type, when Profile checks are performed.**

|        | ios | watchos | tvos | osx* |
|--------|-----|---------|------|------|
| iphone | ✅   | ✅       | ❌    | ❌    |
| ipod   | ✅   | ✅       | ❌    | ❌    |
| ipad   | ✅   | ✅       | ❌    | ❌    |
| watch  | ✅   | ✅       | ❌    | ❌    |
| tvOS   | ❌   | ❌       | ✅    | ❌    |

*macOS projects are not supported by the step

- Improve `fetch_devices`, `result_string` and `run_and_handle_portal_function` function names.
- Fix empty error in case of a failed device registration.
- `rubocop --auto-gen-config` performed to omit new lint issues.
- `test_xcode_managed` is temporary turned off if Xcode version > 13
  - Recent Xcode 13 beta introduced a new behavior: in the case of Xcode managed codesigning Xcode requires the new Apple developer certificates (Apple Development and Apple distribution), regardless of the project's Signing Identity is set to `iPhone Developer/iPhone Distribution` or `Apple Development/Apple Distribution`. This will be fixed in a further PR.

### CODE REVIEW

**Review by Bence1001 — APPROVED** [review]
LGTM! 👌

File: bitrise.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ workflows:
         - content: |-
             #!/bin/bash
             set -e
+            gem install bundler -v 1.17.3 --force
```
Comment (Bence1001) [inline bitrise.yml:13]: Why is this needed?

**Review by Bence1001 — APPROVED** [review]
LGTM! I have one small question.

File: bitrise.yml (hunk 1 above)
Comment (godrei) [reply bitrise.yml:13]: Gemfile.lock describes the [required bundler version](https://github.com/bitrise-steplib/steps-ios-auto-provision/blob/STEP-6-filter-devices-for-platform/Gemfile.lock#L242-L243) (1.17.3), previously this version was preinstalled on the VM we used for testing.

This has been changed, the new preinstalled bundler version is 2.2.24: https://github.com/bitrise-io/bitrise.io/blob/master/system_reports/linux-docker-android-20.04.log#L404

**Review by Bence1001 — APPROVED** [review]
